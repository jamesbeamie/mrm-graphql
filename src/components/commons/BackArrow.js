import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import '../../assets/styles/backArrow.scss';

/**
 * Reusable component for a back arrow
 *
 * @param {object} redirectUri
 *
 * @returns {JSX}
 */

/* eslint-disable jsx-a11y/accessible-emoji */
const BackArrow = ({ redirectUri }) => (
  <NavLink to={redirectUri} className="arrow">
    <span className="arrow-symbol">&#11013;</span>
    <span>Back</span>
  </NavLink>
);

BackArrow.propTypes = {
  redirectUri: PropTypes.string.isRequired,
};

export default BackArrow;
