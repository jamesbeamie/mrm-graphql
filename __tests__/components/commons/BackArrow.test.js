import React from 'react';
import { shallow } from 'enzyme';
import BackArrow from '../../../src/components/commons/BackArrow';

const initProps = {
  redirectUri: '',
};

describe('Back Arrow Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<BackArrow {...initProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
