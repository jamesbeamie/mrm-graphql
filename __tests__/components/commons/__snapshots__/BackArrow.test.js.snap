// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back Arrow Component should render correctly 1`] = `
<NavLink
  activeClassName="active"
  ariaCurrent="true"
  className="arrow"
  to=""
>
  <span
    className="arrow-symbol"
  >
    â¬…
  </span>
  <span>
    Back
  </span>
</NavLink>
`;
